### PR TITLE
Adding command line tests for project.json restore scenarios with prerelease versions

### PR DESCRIPTION
**Pull request by emgarten**
This adds tests for https://github.com/NuGet/Home/issues/1304 and other resolver scenarios involving stable and prerelease packages.

//cc @deepakaravindr @feiling @yishaigalatzer @MeniZalzman @zhili1208

### CODE REVIEW

**Comment by zhili1208** [comment]
looks good to me

File: test/NuGet.CommandLine.Test/RestoreProjectJsonTest.cs [hunk 1]
```diff
@@ -9,6 +10,418 @@ namespace NuGet.CommandLine.Test
     public class RestoreProjectJsonTest : IDisposable
     {
         [Fact]
+        public void RestoreProjectJson_FloatReleaseLabelHighestPrelease()
+        {
+            // Arrange
+            var tempPath = Path.GetTempPath();
+            var workingPath = Path.Combine(tempPath, Guid.NewGuid().ToString());
+            var repositoryPath = Path.Combine(workingPath, Guid.NewGuid().ToString());
+            var currentDirectory = Directory.GetCurrentDirectory();
```
Comment (deepakaravindr) [inline test/NuGet.CommandLine.Test/RestoreProjectJsonTest.cs:19]: You may simplify these calls with the following lines

var workingPath =  TestFileSystemUtility.CreateRandomTestFolder();
var repositoryPath = TestFileSystemUtility.CreateRandomTestFolder();

File: test/NuGet.CommandLine.Test/RestoreProjectJsonTest.cs [hunk 2]
```diff
@@ -9,6 +10,418 @@ namespace NuGet.CommandLine.Test
...
+            var nugetexe = Util.GetNuGetExePath();
+
+            _dirs.TryAdd(workingPath, false);
+
+            Util.CreateDirectory(workingPath);
+            Util.CreateDirectory(repositoryPath);
+            Util.CreateDirectory(Path.Combine(workingPath, ".nuget"));
+            Util.CreateTestPackage("packageA", "1.0.0", repositoryPath, "win8", "packageC", "1.0.0");
+            Util.CreateTestPackage("packageB", "1.0.0", repositoryPath, "win8", "packageC", "[2.1.0]");
```
Comment (deepakaravindr) [inline test/NuGet.CommandLine.Test/RestoreProjectJsonTest.cs:322]: Not sure if I missed it. But, do you want to add a test with a version constraint [4.0.0, ) like in the bug?

**Comment by deepakaravindr** [comment]
:shipit:

File: test/NuGet.CommandLine.Test/RestoreProjectJsonTest.cs (hunk 1 above)
Comment (emgarten) [inline test/NuGet.CommandLine.Test/RestoreProjectJsonTest.cs:19]: I think it's easier to debug when they are in the same working folder, but either would work.

File: test/NuGet.CommandLine.Test/RestoreProjectJsonTest.cs (hunk 2 above)
Comment (emgarten) [inline test/NuGet.CommandLine.Test/RestoreProjectJsonTest.cs:322]: There is a functional test that closely matches the bug in nuget3 commands.

**Pull request closed without merging**